File: 1001-1250/1200. Minimum Absolute Difference/Solution.java
import java.util.* ; 

class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        Arrays.sort( arr ) ; 
        List<List<Integer>> ans = new ArrayList<>() ; 

        int diff = Integer.MAX_VALUE ; 
        for( int i=1 ; i<arr.length ; ++i ){
            diff = Math.min( diff , arr[i] - arr[i-1] ) ; 
        }

        for( int i=1 ; i<arr.length ; ++i ){
            if( arr[i] - arr[i-1] == diff ){
                List<Integer> curr = new ArrayList<>() ; 
                curr.add( arr[i-1] ) ; 
                curr.add( arr[i] ) ; 
                ans.add( curr) ; 
            }
        }

        return ans ; 
    }
}

File: 1001-1250/1200. Minimum Absolute Difference/solution.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        ans = [] 
        arr.sort() 
        diff = float( 'inf' )
        for i in range( 1 , len(arr) ) : 
           curr = arr[i] - arr[i-1]
           if curr == diff : 
                ans.append( [ arr[i-1] , arr[i] ] )
           elif curr < diff : 
                diff = curr
                ans = [[arr[i-1],arr[i]]]

        return ans 